common4j/src/test/com/microsoft/identity/common/java/crypto/StorageEncryptionManagerTest.java

```diff
@@ -0,0 +1,246 @@
+    }
+
+    @Test
+    public void testIsEncryptedByThisKeyIdentifier_True() throws ClientException {
+       Assert.assertTrue(
+               StorageEncryptionManager.isEncryptedByThisKeyIdentifier(
+                       TEXT_ENCRYPTED_BY_PREDEFINED_KEY,
+                       PREDEFINED_KEY_IDENTIFIER));
+
+    }
+
+    @Test
+    public void testIsEncryptedByThisKeyIdentifier_False() throws ClientException {
+        Assert.assertFalse(
+                StorageEncryptionManager.isEncryptedByThisKeyIdentifier(
+                        TEXT_ENCRYPTED_BY_PREDEFINED_KEY,
+                        ANDROID_WRAPPED_KEY_IDENTIFIER));
+    }
+
+    @Test
+    public void testIsEncryptedByThisKeyIdentifier_StringNotEncrypted() throws ClientException {
+        Assert.assertFalse(
+                StorageEncryptionManager.isEncryptedByThisKeyIdentifier(
+                        TEXT_TO_BE_ENCRYPTED_WITH_PREDEFINED_KEY,
+                        PREDEFINED_KEY_IDENTIFIER));
+    }
+
+    @Test
+    public void testIsEncryptedByThisKeyIdentifier_StringNotProperlyEncrypted() {
+        Assert.assertFalse(
+                StorageEncryptionManager.isEncryptedByThisKeyIdentifier(
+                        "cE1".getBytes(ENCODING_UTF8),
+                        PREDEFINED_KEY_IDENTIFIER));
+
+        Assert.assertFalse(
+                StorageEncryptionManager.isEncryptedByThisKeyIdentifier(
+                        "c".getBytes(ENCODING_UTF8),
+                        PREDEFINED_KEY_IDENTIFIER));
+    }
+
+    @Test
+    public void testIsEncryptedByThisKeyIdentifier_EncodeVersionNotSupported() {
+        Assert.assertFalse(
+                StorageEncryptionManager.isEncryptedByThisKeyIdentifier(
+                        EXPECTED_ENCRYPTED_TEXT_1_WITH_MALFORMED_ENCODE_VERSION,
+                        PREDEFINED_KEY_IDENTIFIER));
+    }
+
+    @Test
+    public void testDecryptedTruncatedString() {
+        try {
+            final StorageEncryptionManager manager = new MockStorageEncryptionManager(PREDEFINED_KEY_IV, new MockAES256KeyLoader(PREDEFINED_KEY, PREDEFINED_KEY_IDENTIFIER));
+            final byte[] encryptedByteArray = TEXT_ENCRYPTED_BY_PREDEFINED_KEY;
+            final byte[] truncatedByteArray = Arrays.copyOf(encryptedByteArray, encryptedByteArray.length / 2);
+            manager.decrypt(truncatedByteArray);
+            Assert.fail();
+        } catch (final ClientException e){
+            Assert.assertEquals(e.getErrorCode(), ErrorStrings.DECRYPTION_FAILED);
+            Assert.assertEquals(((ClientException)e.getSuppressedException().get(0)).getErrorCode(), HMAC_MISMATCH);
+        }
+
+        try {
+            final StorageEncryptionManager manager = new MockStorageEncryptionManager(PREDEFINED_KEY_IV, new MockAES256KeyLoader(PREDEFINED_KEY, PREDEFINED_KEY_IDENTIFIER));
+            manager.decrypt(new String(TEXT_ENCRYPTED_BY_PREDEFINED_KEY, ENCODING_UTF8).substring(0, 25).getBytes(ENCODING_UTF8));
+            Assert.fail();
```

Comment:
Can you throw an AssertionException here so we get a stack trace? (Similar comment for all Assert.fail())

Reply:
AssertionException doesn't exist. 

Stack trace here won't help much though - since it's not reflecting the actual error.

Reply:
It is called as `AssertionError`, but yeah it is only helpful if you have the original exception then you can do `throw new AssertionError(e)`, otherwise if we don't have that then `throw new AssertionError()` and `Assert.fail()` should be equivalent.